lib/wallet/src/Cardano/Wallet/DB/Pure/Implementation.hs

```diff
@@ -521,10 +524,19 @@ filterTxHistory minWithdrawal order range address =
     filterWithdrawals = maybe
         (const True)
         (\inf -> atLeast inf . withdrawals . fst)
+
     addressInTxOut addr = any (\(TxOut addr' _) -> addr' == addr)
-    filterAddress = maybe
-        (const True)
-        (\addr -> addressInTxOut addr . outputs . fst)
+    checkOut addr = addressInTxOut addr . outputs . fst
+
+    isAddressPresent addr = \case
+        (_, Just (TxOut addr' _)) -> addr' == addr
+        _ -> False
+    addressInInp addr = any (isAddressPresent addr)
+    checkInp addr = addressInInp addr . resolvedInputs . fst
+
+    filterAddress addrM txhistory = TR.trace ("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!txhistory: "<> show txhistory) $ case addrM of
```

Comment:
trace leftover ?

Reply:
it was the case for some commit, but not in the latests one